node/src/remote.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import axios from 'axios'
+
+import { type Table, tableFromIPC } from 'apache-arrow'
+
+// Immutable data object
+export class RemoteQuery {
+  constructor (
+    readonly vector: number[],
+    readonly k: number,
+    readonly nprobes: number,
+    readonly refine_factor?: number,
+    readonly columns?: string[],
+    readonly filter?: string
+  ) {}
+}
+
+export class HttpLancedbClient {
+  private readonly _url: string
+
+  public constructor (url: string) {
+    this._url = url
+  }
+
+  get uri (): string {
+    return this._url
+  }
+
+  public async search (
+    tableName: string,
+    query: RemoteQuery
+  ): Promise<Table<any>> {
+    const response = await axios.post(
+            `${this._url}/table/${tableName}/`,
```

Comment:
This might be just tmp solution, i.e., how to handle multi-users/tenant using this URL pattern?